### PR TITLE
adds typography and colors to the design system

### PR DESCRIPTION
**Pull request by alexlbr**
- Adds typography (fontFamily, fontSize, fontWeight, lineHeight, letterSpacing, textAlign, and fontStyle.) to the design system.
- Updates the H1, H2, ... H5 typography to use the ones from the design system
- Renames some of the color consts to make them more semantic.
- Adds colors to the design system
- Updates the Header to use the "white" color from the design system

### CODE REVIEW

File: src/components/layout/Box.js [hunk 1]
```diff
@@ -1,9 +1,31 @@
...
+      {children}
+    </StyledBox>
+  )
+})
+
+Box.displayName = 'Box'
+Box.defaultProps = {
+  fontFamily: 'barlow',
+  fontWeight: 'normal',
+  color: 'darkGrey',
```
Comment (ric9176) [inline src/components/layout/Box.js]: why here do we not use the constant DARK_GREY?

File: src/components/layout/Box.js [hunk 2]
```diff
@@ -1,9 +1,31 @@
...
+  return (
+    <StyledBox {...props} ref={ref}>
+      {children}
+    </StyledBox>
+  )
+})
+
+Box.displayName = 'Box'
+Box.defaultProps = {
+  fontFamily: 'barlow',
```
Comment (ric9176) [inline src/components/layout/Box.js]: Is there not at constant for FONT_FAMILY, or can we not use those here?

File: src/components/layout/Header.js [hunk 3]
```diff
@@ -97,14 +97,18 @@ HeaderSection.defaultProps = {
...
-  padding: 0 !important;
-  margin: 0 !important;
-  font-size: 1.16rem !important;
-  color: ${WHITE};
-  font-weight: normal;
   text-shadow: 1px -1px 17px ${reactBlue(0.4)};
 `
 
+H2Header.defaultProps = {
+  color: 'white',
```
Comment (ric9176) [inline src/components/layout/Header.js]: same questions re contstant for color

Comment (alexlbr) [reply src/components/layout/Header.js]: the value of the default prop color is "white" per this line https://github.com/reactgraphqlacademy/reactgraphqlacademy/blob/ds-fonts/src/config/styles.js#L122. It means the prop needs to match a key (name of a color) of that object. The variable "WHITE" references the value "#fff" https://github.com/reactgraphqlacademy/reactgraphqlacademy/blob/ds-fonts/src/config/styles.js#L20, so we can't use the const WHITE to match the key in the theme cause "#fff" is not the name of a color. I think this is solved nicely with types, as we do in UM with TypeScript. Without types, I don't see a simple solution for this 🤔

**Pull request merged**